lib/CMakeLists.txt

```diff
@@ -0,0 +1,171 @@
+
+# Read external project versions
+file(STRINGS ${CMAKE_CURRENT_SOURCE_DIR}/VERSIONS VERSIONS_LIST)
+foreach(ITEM ${VERSIONS_LIST})
+    string(REPLACE "=" ";" ARGS ${ITEM})
+    list(GET ARGS 0 LIB)
+    list(GET ARGS 1 VERSION)
+    set(${LIB} ${VERSION})
+endforeach()
+
+set(VIPS_VERSION ${VIPS_VERSION} PARENT_SCOPE)
+string(REPLACE "." "_" EXIF_VERSION_TAG ${EXIF_VERSION})
+
+# Set the host target
+if (NOT DEFINED BUILD_TARGET)
+  set(CONFIGURE_HOST "--host=x86_64-pc-linux")
+elseif(${BUILD_TARGET} STREQUAL "w64")
+  set(CONFIGURE_HOST "--host=x86_64-w64-mingw32")
+endif()
+
+if (NOT DEFINED CMAKE_BUILD_TYPE)
+set(CONFIGURE_CFLAGS "${CMAKE_C_FLAGS}")
+set(CONFIGURE_CXXFLAGS "${CMAKE_CXX_FLAGS}")
+elseif (${CMAKE_BUILD_TYPE} STREQUAL "Release")
+  set(CONFIGURE_CFLAGS ${CMAKE_C_FLAGS_RELEASE})
+  set(CONFIGURE_CXXFLAGS ${CMAKE_CXX_FLAGS_RELEASE})
+elseif (${CMAKE_BUILD_TYPE} STREQUAL "Debug")
+  set(CONFIGURE_CFLAGS ${CMAKE_C_FLAGS_DEBUG})
+  set(CONFIGURE_CXXFLAGS ${CMAKE_CXX_FLAGS_DEBUG})
+endif()
+
+# Workaround to set more than one compiler flags
+string(REPLACE " " "\ " CONFIGURE_CFLAGS "${CONFIGURE_CFLAGS}")
+string(REPLACE " " "\ " CONFIGURE_CXXFLAGS "${CONFIGURE_CXXFLAGS}")
+
+list(APPEND CONFIGURE_VARS
+  PKG_CONFIG_PATH=${EXT_INSTALL_DIR}/lib/pkgconfig:$PKG_CONFIG_PATH
+  CFLAGS=${CONFIGURE_CFLAGS}
+  CXXFLAGS=${CONFIGURE_CXXFLAGS}
+  LDFLAGS=-L${EXT_INSTALL_DIR}/lib,-Wl,-rpath=${EXT_INSTALL_DIR}/lib
+  CPPFLAGS=-I${EXT_INSTALL_DIR}/include
+  --prefix=${EXT_INSTALL_DIR}
+  ${CONFIGURE_HOST}
+  )
+
+ExternalProject_Add(libimagequant
+  URL "https://github.com/ImageOptim/libimagequant/archive/${LIQ_VERSION}.tar.gz"
+  PREFIX "${CMAKE_CURRENT_BINARY_DIR}/libimagequant"
+  CONFIGURE_COMMAND ./configure CC=${CMAKE_C_COMPILER} CFLAGS=${CONFIGURE_CFLAGS} --prefix=${EXT_INSTALL_DIR} --with-sse --without-openmp
+  BUILD_IN_SOURCE 1
+  )
+
+if(${BUILD_TARGET} STREQUAL "w64")
+  ExternalProject_Add_Step(libimagequant rename-dll
+  COMMAND mv libimagequant.so libimagequant.dll
+  WORKING_DIRECTORY ${EXT_INSTALL_DIR}/lib/
+  DEPENDEES install
+  )
+  ExternalProject_Add_StepTargets(libimagequant rename-dll)
+endif()
+
+ExternalProject_Add(libexif
+  URL "https://github.com/libexif/libexif/releases/download/libexif-${EXIF_VERSION_TAG}-release/libexif-${EXIF_VERSION}.tar.gz"
+  PREFIX "${CMAKE_CURRENT_BINARY_DIR}/libexif"
+  CONFIGURE_COMMAND ${CMAKE_CURRENT_BINARY_DIR}/libexif/src/libexif/configure
+  ${CONFIGURE_VARS}
+  --enable-shared
+  --disable-static
+  --disable-docs
+  )
+
+ExternalProject_Add(libjpeg-turbo
+  URL "https://github.com/libjpeg-turbo/libjpeg-turbo/archive/${JPG_VERSION}.tar.gz"
+  PREFIX "${CMAKE_CURRENT_BINARY_DIR}/libjpeg-turbo"
+  PATCH_COMMAND autoreconf -fiv
+  CONFIGURE_COMMAND ./configure
+  ${CONFIGURE_VARS}
+  --enable-shared
+  --disable-static
+  --with-jpeg8
+  BUILD_IN_SOURCE 1
+  )
+
+ExternalProject_Add(libpng
+  URL "http://prdownloads.sourceforge.net/libpng/libpng-${PNG_VERSION}.tar.xz"
```

Comment:
FAR: Probably not a big deal, but some of these are not the exact same URLs than `build.sh` (different extensions, http instead of https...). Can you fix them in follow-up commit?

Reply:
I will fix that in this one, thank you